From 97d924dfed7284b103f588f009d63ec5891ba347 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Thu, 7 Jul 2016 11:39:03 +0300
Subject: [PATCH 4/6] abuild: support arch="all !armhf"

this allows blacklisting architectures instead of just
whitelisting them. useful when more architectures comeabout.
---
 abuild.in | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 72f8978..49e2630 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1607,7 +1607,16 @@ srcpkg() {
 
 # return true if arch is supported or noarch
 check_arch() {
-	list_has $CARCH $arch || [ "$arch" = "noarch" ] || [ "$arch" = "all" ]
+	local ret=1
+	local i
+	for i in $arch; do
+		case $i in
+		all | noarch) ret=0 ;;
+		"$CARCH") ret=0 ;;
+		"!$CARCH") return 1 ;;
+		esac
+	done
+	return $ret
 }
 
 # return true if libc is not masked in options
-- 
2.9.0

